src/main/java/com/manywho/services/sql/services/DataService.java

```diff
@@ -60,19 +60,42 @@ public DataService(MObjectFactory mObjectFactory, QueryStrService queryStrServic
         }
     }
 
-    public List<MObject> fetchBySearch(TableMetadata tableMetadata, Sql2o sql2o, String sqlSearch, List<Object> placeHolderParameters) throws SQLException {
-        try (Connection con = sql2o.open()) {
-            PreparedStatement preparedStatement = con.getJdbcConnection().prepareStatement(sqlSearch);
+    /**
+     * This method create a query object from library Sql2o with the parameters populated
+     *
+     * we have used com.healthmarketscience.sqlbuilder to build the "where" part, each parameter has been marked as "?",
+     * but we need to use a different format when using sql2o to execute the query, the format is naming the parameters
+     * like this ":paramname"
+     *
+     */
+    public Query createQuery(Connection connection, String sqlSearch, List<Object> placeHolderParameters) {
```

Comment:
I don't think this is a particularly good idea - it might be OK now, but in the future it might encourage others to "just add code in here" to fix some bug, and it might bypass the safety we're getting with parameters.

It looks like Sql2o is in maintenance mode anyway (no release in a few years, and the one version that was released had one beta since 2016), so maybe we should swap it out for something more maintained and supports the standard `?` for parameters so we don't have to modify them - Jdbi is pretty decent, and the API is very similar.

Reply:
I agree, it requires a bit of maintenance work. 
Also we need to check if the libraries are compatible, because both libraries Sql2O and healthmarketscience.sqlbuilder have support for parameters with  `?` but they keep the index internal so it is hard to make them work together.